src/circular_buffer.c

```diff
@@ -44,68 +44,66 @@
 #include <debug.h>
 
 typedef struct _circular_buffer {
-
     uint32_t* buffer;
     uint32_t buffer_size;
-
     uint32_t output;
     uint32_t input;
     uint32_t overflows;
 } _circular_buffer;
 
-// allocated
-//
-// Returns the number of buffer slots currently allocated
-static inline uint32_t _circular_buffer_allocated(circular_buffer buffer) {
-    register uint32_t r = (buffer->input >= buffer->output)? 0:
-            buffer->buffer_size;
-    r += buffer->input - buffer->output;
-    assert(r < buffer->buffer_size);
-    return r;
+// Returns the next highest power of 2 of a value
+static inline uint32_t _next_power_of_two(uint32_t v) {
+    return 1 << (32 - __builtin_clz(v));
```

Comment:
This certainly works - thanks @neworderofjamie!  I assume that this is what you had in mind...